### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by tomaz82**
Fix a couple of warnings on Windows and Ubuntu, mostly deleting unused variables.

### CODE REVIEW

**Comment by joolswills** [comment]
I would prefer to leave the whitespace cleanup for now - we have too many ongoing PRs that it would cause to need rebasing etc.

**Comment by tomaz82** [comment]
Ok, I can undo those, what about the fixed warnings?

**Comment by joolswills** [comment]
Happy to merge them.

**Comment by tomaz82** [comment]
There, now it's just the warnings fixed

File: es-app/src/guis/GuiGamelistFilter.cpp [hunk 1]
```diff
@@ -55,11 +55,10 @@ void GuiGamelistFilter::addFiltersToMenu()
 
 		FilterIndexType type = (*it).type; // type of filter
 		std::map<std::string, int>* allKeys = (*it).allIndexKeys; // all possible filters for this type
-		std::vector<std::string>* allFilteredKeys = (*it).currentFilteredKeys; // current keys being filtered for
+//		std::vector<std::string>* allFilteredKeys = (*it).currentFilteredKeys; // current keys being filtered for
```
Comment (joolswills) [inline es-app/src/guis/GuiGamelistFilter.cpp]: if this line is not needed it would be better to just remove.

File: es-app/src/guis/GuiGamelistFilter.cpp [hunk 2]
```diff
@@ -55,11 +55,10 @@ void GuiGamelistFilter::addFiltersToMenu()
...
 		std::string menuLabel = (*it).menuLabel; // text to show in menu
 		std::shared_ptr< OptionListComponent<std::string> > optionList;
 
-
```
Comment (joolswills) [inline es-app/src/guis/GuiGamelistFilter.cpp]: unrelated whitespace change.

File: es-core/src/components/ImageGridComponent.h [hunk 3]
```diff
@@ -218,7 +218,7 @@ void ImageGridComponent<T>::updateImages()
 	Eigen::Vector2i gridSize = getGridSize();
 
 	int cursorRow = mCursor / gridSize.x();
-	int cursorCol = mCursor % gridSize.x();
+//	int cursorCol = mCursor % gridSize.x();
```
Comment (joolswills) [inline es-core/src/components/ImageGridComponent.h]: as above - please remove if not needed.

File: es-core/src/components/VideoComponent.cpp [hunk 4]
```diff
@@ -27,7 +27,7 @@ void writeSubtitle(const char* gameName, const char* systemName, bool always)
 	int end = (int)(Settings::getInstance()->getInt("ScreenSaverSwapVideoTimeout") / (1000));
 	if (always) {
 		fprintf(file, "1\n00:00:01,000 --> 00:00:");
-		fprintf(file, std::to_string(end).c_str());
+		fprintf(file, "%s", std::to_string(end).c_str());
 		fprintf(file, ",000\n");
```
Comment (joolswills) [inline es-core/src/components/VideoComponent.cpp]: Please can you leave the printfs as @pjft is going to re-do it (this could be a single line for example)

**Comment by tomaz82** [comment]
All your reviews should be fixed now.

**Comment by joolswills** [comment]
thanks.
